go/marshal/encode.go

```diff
@@ -447,18 +448,21 @@ func typeFields(t reflect.Type, parentStructTypes []reflect.Type, options encode
 			index:     i,
 			nomsType:  nt,
 			omitEmpty: tags.omitEmpty,
+			optional:  tags.omitEmpty,
```

Comment:
I thought about this a bit and I'm not sure that re-using the omitEmpty tag for optional is the right thing to do. Better to have a clean break and anybody that wants to migrate from the omitEmpty semantics to the optional semantics can do so themselves.